### PR TITLE
Homework Podstawy C++ password-check

### PR DESCRIPTION
**Pull request by Crazyayane07**
Authors: @Crazyayane07, @krzysztof-kryczka

### CODE REVIEW

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @Crazyayane07

**Comment by ziobron** [comment]
🏅 +1 XP for cooperation goes to @krzysztof-kryczka

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @Crazyayane07

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check-bonus
🏅 Granted 4 XP for @krzysztof-kryczka

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @Crazyayane07

**Comment by ziobron** [comment]
⏱ 3 XP extra for @Crazyayane07 for on-time delivery

**Comment by ziobron** [comment]
✅ CI Passed - Task: password-check
🏅 Granted 8 XP for @krzysztof-kryczka

**Comment by ziobron** [comment]
⏱ 3 XP extra for @krzysztof-kryczka for on-time delivery

File: homework/password-check/validation.cpp [hunk 1]
```diff
@@ -1,2 +1,60 @@
...
+}
+
+ErrorCode checkPasswordRules(const std::string& password) {
+    if (password.length() < 9) {
+        return ErrorCode::PasswordNeedsAtLeastNineCharacters;
+    }
+    if (!(std::any_of(password.begin(), password.end(), ::isdigit))) {
+        return ErrorCode::PasswordNeedsAtLeastOneNumber;
+    }
+    if (password.find_first_not_of("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890") == std::string::npos) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:46]: int isalnum( int ch );

File: homework/password-check/validation.cpp [hunk 2]
```diff
@@ -1,2 +1,60 @@
...
+    bool pass = (password == repeatPassword) ? true : false;
+
+    return pass;
+}
+
+ErrorCode checkPasswordRules(const std::string& password) {
+    if (password.length() < 9) {
+        return ErrorCode::PasswordNeedsAtLeastNineCharacters;
+    }
+    if (!(std::any_of(password.begin(), password.end(), ::isdigit))) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:43]: better us none_of

File: homework/password-check/validation.cpp [hunk 3]
```diff
@@ -1,2 +1,60 @@
...
+        return ErrorCode::PasswordNeedsAtLeastOneNumber;
+    }
+    if (password.find_first_not_of("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ01234567890") == std::string::npos) {
+        return ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter;
+    }
+    if (!(std::any_of(password.begin(), password.end(), isupper))) {
```
Comment (nauka-programowania-MA) [inline homework/password-check/validation.cpp:49]: better us none_of

**Review by nauka-programowania-MA — APPROVED** [review]

**Pull request closed without merging**